src/cargo/sources/registry/index.rs

```diff
@@ -437,11 +437,13 @@ impl<'cfg> RegistryIndex<'cfg> {
     /// checking the integrity of a downloaded package matching the checksum in
     /// the index file, aka [`IndexSummary`].
     pub fn hash(&mut self, pkg: PackageId, load: &mut dyn RegistryData) -> Poll<CargoResult<&str>> {
-        let req = OptVersionReq::exact(pkg.version());
+        let mut req = OptVersionReq::exact(pkg.version());
+        // Since crates.io allows crate versions to differ only by build metadata,
```

Comment:
Because of rust-lang/crates.io#6518, should we change this to "Since some registries may allow"?

Or should we go a step further and make this a requirement on all registries?  If not, should we at least document it as a best practice?

Reply:
I am open to wordsmithing the comment.

I'm pretty sure we already documented it. If not we should.

Reply:
Found it! https://doc.rust-lang.org/cargo/reference/resolver.html?highlight=metadata#version-metadata

Reply:
In light of everything, maybe this should be "Since some registries may incorrectly allow ..."

Reply:
(plus the other comment)